int32 num_robot
int32 data
